File: Java/civitas/src/GUI/Controlador.java
package GUI;
import civitas.CivitasJuego;
import civitas.GestionesInmobiliarias;
import civitas.OperacionInmobiliaria;
import civitas.Jugador;
import civitas.OperacionesJuego;
import civitas.SalidasCarcel;
import java.util.ArrayList;

public class Controlador {
    private CivitasJuego juego;
    private CivitasView vista;
    
    public Controlador(CivitasJuego juego, CivitasView vista) {
        this.juego = juego;
        this.vista = vista;
    }
    
    void juega(){
        vista.setCivitasJuego(juego);
        while(juego.finalDelJuego()==false){
            vista.actualizarVista();
            OperacionesJuego siguiente = juego.siguientePaso();
            if(siguiente != OperacionesJuego.PASAR_TURNO){
                vista.mostrarEventos();
            }
            boolean fin = juego.finalDelJuego();
            if(fin == false){
                if(siguiente==OperacionesJuego.COMPRAR){
                        Respuestas respuesta = vista.comprar();
                        if(respuesta == Respuestas.SI){
                            juego.comprar();
                            juego.siguientePasoCompletado(OperacionesJuego.COMPRAR);
                        }
                        else{
                            juego.siguientePasoCompletado(OperacionesJuego.COMPRAR);
                        }
                }
                if(siguiente==OperacionesJuego.GESTIONAR){
                    vista.gestionar();
                    int ges = vista.getGestion();
                    int pro = vista.getPropiedad();
                    OperacionInmobiliaria operacionInmo = new OperacionInmobiliaria(pro, GestionesInmobiliarias.values()[ges]);
                    if(GestionesInmobiliarias.CANCELAR_HIPOTECA == operacionInmo.getGestion()){
                        juego.cancelarHipoteca(operacionInmo.getNumPropiedad());
                    }
                    if(GestionesInmobiliarias.CONSTRUIR_CASA == operacionInmo.getGestion()){
                        juego.construirCasa(operacionInmo.getNumPropiedad());
                    }
                    if(GestionesInmobiliarias.CONSTRUIR_HOTEL == operacionInmo.getGestion()){
                        juego.construirHotel(operacionInmo.getNumPropiedad());
                    }
                    if(GestionesInmobiliarias.HIPOTECAR == operacionInmo.getGestion()){
                        juego.hipotecar(operacionInmo.getNumPropiedad());
                    }
                    if(GestionesInmobiliarias.VENDER == operacionInmo.getGestion()){
                        juego.vender(operacionInmo.getNumPropiedad());
                    }
                    if(GestionesInmobiliarias.TERMINAR == operacionInmo.getGestion()){
                        juego.siguientePasoCompletado(OperacionesJuego.GESTIONAR);
                    }
                }
                if(siguiente == OperacionesJuego.SALIR_CARCEL){
                    SalidasCarcel salida = vista.salirCarcel();
                    if(salida == SalidasCarcel.PAGANDO){
                        juego.salirCarcelPagando();
                    }
                    else{
                        juego.salirCarcelTirando();
                    }
                    juego.siguientePasoCompletado(OperacionesJuego.SALIR_CARCEL);
                }
            }
            System.out.println("********************************************\n");
        }
        juego.ranking();
    }
}

File: Ruby/civitas/pkg/civitas-0.0.1/lib/controlador.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require_relative 'civitasjuego'
require_relative 'vista_textual'
require_relative 'operaciones_juego'
require_relative 'operaciones_inmobiliarias'
require_relative 'salidas_carcel'
require_relative 'respuestas'
 module Civitas
  class Controlador
    def initialize(_juego,_vista)
        @juego = _juego
        @vista = _vista
    end
    
    def juega()
      @vista.setCivitasJuego(@juego)
      while(@juego.final_del_juego()==false)
        @vista.actualizarVista()
        @vista.pausa()
        siguiente = @juego.siguiente_paso()
        @vista.mostrarSiguienteOperacion(siguiente)
        
        if(siguiente != Operaciones_juego::PASAR_TURNO)
          @vista.mostrarEventos()
          puts "No es pasar Turno"
        end
        fin = @juego.final_del_juego()
        if(fin==false)
          puts "fin==false"
          if(siguiente == Operaciones_juego::COMPRAR)
            puts "Siguiente operacion igual a false"
              respuesta = @vista.comprar()
              if(respuesta == Respuestas::SI)
                puts "respuesta es SI"
                @juego.comprar()
                @juego.siguiente_paso_completado(Operaciones_juego::COMPRAR)
              else
                puts "Respuesta es NO"
                @juego.siguiente_paso_completado(Operaciones_juego::COMPRAR)
              end
          end
            
          if(siguiente ==Operaciones_juego::GESTIONAR)
            puts "If de GESTIONAR"
            @vista.gestionar()
            
            operacioninmo = Operaciones_inmobiliarias.new(@vista.get_gestion(), @vista.get_propiedad())
            if(Gestiones_inmobiliarias::CANCELAR_HIPOTECA == @vista.get_gestion())
              @juego.cancelar_hipoteca(@vista.get_propiedad())
              
            end
            if(Gestiones_inmobiliarias::CONSTRUIR_CASA==@vista.get_gestion())
              @juego.construir_casa(@vista.get_propiedad())
              
            end
            if(Gestiones_inmobiliarias::CONSTRUIR_HOTEL==@vista.get_gestion())
              @juego.construir_hotel(@vista.get_propiedad())
              
            end
            if(Gestiones_inmobiliarias::HIPOTECAR==@vista.get_gestion())
              @juego.hipotecar(@vista.get_propiedad())
              
            end
            if(Gestiones_inmobiliarias::VENDER==@vista.get_gestion())
              @juego.vender(@vista.get_propiedad())
              
            end
            if(Gestiones_inmobiliarias::TERMINAR==@vista.get_gestion())
              @juego.siguiente_paso_completado(Operaciones_juego::GESTIONAR)
              
            end
          end
          if(siguiente == Operaciones_juego::SALIR_CARCEL)
            salida = @vista.salir_carcel()
            if(salida == Salidas_carcel::PAGANDO)
              @juego.salir_carcel_pagando()
            else
              @juego.salir_carcel_tirando()
            end
            @juego.siguiente_paso_completado(Operaciones_juego::SALIR_CARCEL)
          end
        end
        puts "SIGUIENTE VALE " + siguiente.to_s()
        puts "*******************************************************************************"
      end
      puts "Voy a ir a ranking"
      @juego.ranking()
    end
  end
end